src/python/pants/backend/python/util_rules/interpreter_constraints_test.py

```diff
@@ -398,3 +398,13 @@ def test_contains(candidate, target, matches) -> None:
         )
         == matches
     )
+
+
+def test_constraints_are_correctly_sorted_at_construction() -> None:
+    # #12578: This list itself is out of order, and `CPython>=3.6,<4,!=3.7.*` is specified with out-of-order component requirements.
```

Comment:
Looks like this might be over 100 lines, which we try to keep under. Black will not fix those for you because they're comments.